include/ideep/operators/matmul.hpp

```diff
@@ -818,23 +819,29 @@ struct matmul_forward : public dnnl::matmul,
     // introduces *an extra reorder* afterwards. Here we keep the weight format
     // untouched thanks to optimizations for both plain and transposed formats
     // in DNNL.
-    IDEEP_ENFORCE(weights.get_data_type() == data_type::f32 ||
-                  weights.get_data_type() == data_type::bf16 ||
-                  weights.get_data_type() == data_type::f16,
-                  "Incorrect data type in weights");
-    dst_data_type = src.get_data_type() == data_type::bf16 ?
-                    data_type::bf16 :
-                    ((src.get_data_type() == data_type::f16) ?
-                      data_type::f16 : data_type::f32);
-    src_desc = src.get_desc().to_type(dst_data_type);
-
+    IDEEP_ENFORCE(
+        weights.get_data_type() == data_type::f32 ||
+            weights.get_data_type() == data_type::bf16 ||
+            weights.get_data_type() == data_type::f16,
+        "Incorrect data type in weights");
+
+    dst_data_type = src.get_data_type() == data_type::bf16
+        ? ((dst.get_data_type() == data_type::f32) ? data_type::f32
+                                                   : data_type::bf16)
+        : ((src.get_data_type() == data_type::f16) ? data_type::f16
+                                                   : data_type::f32);
+    src_desc = (src.get_data_type() == data_type::bf16 &&
+                dst_data_type == data_type::f32)
+        ? src.get_desc()
+        : src.get_desc().to_type(dst_data_type);
 #ifdef __aarch64__
     // for aarch64 ACL backend with fixed format kernels, the weights are
     // always in blocked layout, so, set the descriptor to tag::any for the backend
     // to decide the format
-    weights_desc = tensor::desc(weights.get_dims(), dst_data_type, tag::any);
+    wei_data_type = (src.get_data_type() == data_type::bf16 &&
+                   dst_data_type == data_type::f32) ? data_type::bf16 : dst_data_type;
```

Comment:
Again, why do we set different data type for weights for aarch64?

Reply:
Hi @jgong5 , the logic updated now. I was testing it for x86 and the change got skipped in the commit!